gapis/resolve/dependencygraph2/graph_visualization/graph_visualization.go

```diff
@@ -16,17 +16,208 @@ package graph_visualization
 
 import (
 	"context"
+	"fmt"
 	"github.com/google/gapid/core/math/interval"
 	"github.com/google/gapid/gapis/api"
 	"github.com/google/gapid/gapis/capture"
 	"github.com/google/gapid/gapis/resolve/dependencygraph2"
 )
 
-func GetGraphVisualizationFromCapture(ctx context.Context, p *capture.Capture) ([]byte, error) {
+const (
+	VK_BEGIN_COMMAND_BUFFER    = "vkBeginCommandBuffer"
+	VK_CMD_BEGIN_RENDER_PASS   = "vkCmdBeginRenderPass"
+	VK_CMD_NEXT_SUBPASS        = "vkCmdNextSubpass"
+	VK_COMMAND_BUFFER          = "vkCommandBuffer"
+	VK_RENDER_PASS             = "vkRenderPass"
+	VK_SUBPASS                 = "vkSubpass"
+	VK_END_COMMAND_BUFFER      = "vkEndCommandBuffer"
+	VK_CMD_END_RENDER_PASS     = "vkCmdEndRenderPass"
+	VK_CMD_DRAW_INDEXED        = "vkCmdDrawIndexed"
+	VK_CMD_DRAW                = "vkCmdDraw"
+	COMMAND_BUFFER             = "commandBuffer"
+	MAXIMUM_LEVEL_IN_HIERARCHY = 10
+)
+
+var (
+	beginCommands = map[string]int{
+		VK_BEGIN_COMMAND_BUFFER:  0,
+		VK_CMD_BEGIN_RENDER_PASS: 1,
+		VK_CMD_NEXT_SUBPASS:      2,
```

Comment:
It looks like these hard-coded levels will fail in the case of `vkCmdExecuteCommands`.

Reply:
Yes, you are right. In this commit I was not considering that problem, but in the next commits it is fixed, also it is defined a getSubCommandLabel function to address this problem.